src/protocol/ipa/mod.rs

```diff
@@ -314,9 +309,22 @@ where
     .unwrap();
 
     let converted_mk_shares = combine_slices(&converted_mk_shares, MK::BITS);
+    let arithmetic_mk_shares: Vec<Replicated<F>> = converted_mk_shares
+        .map(|bitwise_shares| {
+            bitwise_shares
+                .into_iter()
+                .enumerate()
+                .fold(Replicated::ZERO, |acc, (i, bit)| {
+                    // TODO: use try_from to ensure the code fatals if we are trying to use too small of a field
+                    // to hold the match key. This requires an even larger prime field than is currently defined
```

Comment:
Yeah, I could do that.

It's a bit wasteful since the equality check looks at ALL the bits of `r` to check if they are equal to the revealed bits of `c`, so we will be multiplying 64 shares together, not 40.